### PR TITLE
[labs/virtualizer] Fix virtualize directive update to match lit-virtualizer update.  Added test helpers, fixes and scenarios to cover.

### LINKED ISSUE #3052: [labs/virtualizer] `virtualize` directive only re-renders children if `items` or `renderItem` have changed
# Description

When using the `virtualize` directive, children fail to re-render in response to certain types of data changes.

Specifically, children are only re-rendered if data within the `items` array changes (or if the `renderItem` template itself is replaced). This means that any changes to data stored _outside_ the `items` array will fail to trigger re-rendering, even if logic or bindings in the `renderItem` template refer to the changed data.

# Steps to Reproduce

1. Go to the demo link
2. Click "update selected" button
3. See that selected changes in the second list, but not in the first. 

## Live Reproduction Link

[Demo](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcbmltcG9ydCB7dmlydHVhbGl6ZX0gZnJvbSAnQGxpdC1sYWJzL3ZpcnR1YWxpemVyL3ZpcnR1YWxpemUuanMnO1xuXG5AY3VzdG9tRWxlbWVudCgnc2ltcGxlLWdyZWV0aW5nJylcbmV4cG9ydCBjbGFzcyBTaW1wbGVHcmVldGluZyBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgICAgOmhvc3QgeyBcbiAgICAgICAgZGlzcGxheTogYmxvY2s7XG4gICAgICAgIGhlaWdodDogMjAwcHg7XG4gICAgICB9YDtcblxuICBAcHJvcGVydHkoKSBcbiAgcHVibGljIHNlbGVjdGVkID0gbmV3IFNldChbMiwgNV0pO1xuICBcbiAgQHByb3BlcnR5KClcbiAgcHVibGljIGl0ZW1zID0gQXJyYXkuZnJvbShBcnJheSgxMDApLmtleXMoKSk7XG4gIFxuICByZW5kZXIoKSB7XG4gICAgcmV0dXJuIGh0bWxgXG4gICAgIDxkaXY-XG4gICAgICAke3ZpcnR1YWxpemUoe1xuICAgICAgICBzY3JvbGxlcjogdHJ1ZSwgXG4gICAgICAgIGl0ZW1zOiB0aGlzLml0ZW1zLCBcbiAgICAgICAgcmVuZGVySXRlbTogKGl0ZW0pID0-IGh0bWxgPGRpdj4ke2l0ZW19ICR7dGhpcy5zZWxlY3RlZC5oYXMoaXRlbSkgPyAnc2VsZWN0ZWQnIDogJyd9PC9kaXY-YFxuICAgICAgfSl9XG4gICAgIDwvZGl2PlxuICAgIGA7XG4gIH1cbn0ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGhlYWQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vc2ltcGxlLWdyZWV0aW5nLmpzXCI-PC9zY3JpcHQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vb3RoZXItZ3JlZXRpbmcuanNcIj48L3NjcmlwdD5cbiAgXG48L2hlYWQ-XG48Ym9keT5cbiAgPHNpbXBsZS1ncmVldGluZz48L3NpbXBsZS1ncmVldGluZz5cbiAgPG90aGVyLWdyZWV0aW5nPjwvb3RoZXItZ3JlZXRpbmc-XG4gIDxidXR0b24-VXBkYXRlIGBzZWxlY3RlZGA8L2J1dHRvbj5cbiAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCI-XG4gICAgbGV0IHVwZGF0ZXMgPSAwOyBcbiAgICBjb25zdCBidXR0b24gPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yKCdidXR0b24nKTtcbiAgICBjb25zdCBzaW1wbGVHcmVldGluZyA9IGRvY3VtZW50LnF1ZXJ5U2VsZWN0b3IoJ3NpbXBsZS1ncmVldGluZycpO1xuICAgIGNvbnN0IG90aGVyR3JlZXRpbmcgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yKCdvdGhlci1ncmVldGluZycpO1xuXG4gICAgYnV0dG9uLmFkZEV2ZW50TGlzdGVuZXIoJ2NsaWNrJywgKCkgPT4ge1xuICAgICAgY29uc3Qgc2VsZWN0ZWQgPSB1cGRhdGVzJTIgPT0gMCA_IG5ldyBTZXQgKFsxLCA3XSkgOiBuZXcgU2V0IChbMCwgM10pO1xuICAgICAgdXBkYXRlcyArPSAxO1xuICAgICAgc2ltcGxlR3JlZXRpbmcuc2VsZWN0ZWQgPSBzZWxlY3RlZDtcbiAgICAgIG90aGVyR3JlZXRpbmcuc2VsZWN0ZWQgPSBzZWxlY3RlZDtcblxuICAgIH0pO1xuICA8L3NjcmlwdD5cbjwvYm9keT4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4xLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeMy4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX0seyJuYW1lIjoib3RoZXItZ3JlZXRpbmcudHMiLCJjb250ZW50IjoiaW1wb3J0IHtodG1sLCBjc3MsIExpdEVsZW1lbnR9IGZyb20gJ2xpdCc7XG5pbXBvcnQge2N1c3RvbUVsZW1lbnQsIHByb3BlcnR5fSBmcm9tICdsaXQvZGVjb3JhdG9ycy5qcyc7XG5pbXBvcnQgJ0BsaXQtbGFicy92aXJ0dWFsaXplcic7XG5cbkBjdXN0b21FbGVtZW50KCdvdGhlci1ncmVldGluZycpXG5leHBvcnQgY2xhc3MgT3RoZXJHcmVldGluZyBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYFxuICAgICAgOmhvc3QgeyBcbiAgICAgICAgZGlzcGxheTogYmxvY2s7XG4gICAgICAgIGhlaWdodDogMjAwcHg7XG4gICAgICB9YDtcblxuICAgIEBwcm9wZXJ0eSgpIFxuICBwdWJsaWMgc2VsZWN0ZWQgPSBuZXcgU2V0KFsyLCA1XSk7XG4gIFxuICBAcHJvcGVydHkoKVxuICBwdWJsaWMgaXRlbXMgPSBBcnJheS5mcm9tKEFycmF5KDEwMCkua2V5cygpKTtcbiAgXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgPGxpdC12aXJ0dWFsaXplclxuICAgICAgICBzY3JvbGxlciBcbiAgICAgICAgLml0ZW1zPSR7dGhpcy5pdGVtc31cbiAgICAgICAgLnJlbmRlckl0ZW09JHsoaXRlbSkgPT4gaHRtbGA8ZGl2Pk90aGVyICR7aXRlbX0gJHt0aGlzLnNlbGVjdGVkLmhhcyhpdGVtKSA_ICdzZWxlY3RlZCcgOiAnJ308L2Rpdj5gfVxuICAgID5cbiAgICAgPC9saXQtdmlydHVhbGl6ZXI-XG4gICAgYDtcbiAgfVxufSJ9XQ)

## Expected Results

`selected` should change in both lists. 

cc: @Westbrook

### PR DESCRIPTION
**Pull request by usergenic**
* Updated virtualize directive update method to behave as lit-virtualizer element and render when non-item data changes, instead of returning noChange.  Fixes #3052 

* Included `@open-wc/testing` and migrated to `fixture` instead of manual container element creation/insertion/removal. 
* Added a regression test scenario to try and prove behavior parity between virtualize directive and lit-virtualizer.
* Defined some helpers to make analyzing html/text content in tests more succinct and readable.
* Removed a few unnecessary checks and types in tests.

### CODE REVIEW

File: .changeset/yellow-dots-fix.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lit-labs/virtualizer': patch
+---
+
+Changed the update behavior of virtualize directive to match lit-virtualizer's rendering of non-item changes.
```
Comment (graynorton) [inline .changeset/yellow-dots-fix.md:5]: Suggest that we rephrase this to something more like:

The `virtualize` directive will now correctly re-render children when data stored outside the `items` array has changed.

**Review by graynorton — CHANGES_REQUESTED** [review]
Looks good! Just a few minor requests.

File: packages/labs/virtualizer/src/test/scenarios/element-and-directive-parity.test.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    expect(customElements.get('using-virtualize-directive')).to.eq(
+      UsingVirtualizeDirective
+    );
+  });
+});
+
+describe('lit-virtualizer and virtualize directive', () => {
+  ignoreBenignErrors(beforeEach, afterEach);
+
+  it('both render changes based on non-item data changes', async () => {
```
[the remark this answers is not in the document]
Comment (usergenic) [reply packages/labs/virtualizer/src/test/scenarios/element-and-directive-parity.test.ts:71]: Yeah I think that makes sense.  Added.

File: packages/labs/virtualizer/src/test/scenarios/smoke.test.ts [hunk 3]
```diff
@@ -4,57 +4,50 @@
...
+import {virtualize} from '../../virtualize.js';
+import {expect, fixture, html} from '@open-wc/testing';
 
 describe('smoke test', () => {
   ignoreBenignErrors(beforeEach, afterEach);
 
   describe('<lit-virtualizer>', function () {
-    it('registers lit-virtualizer as a custom element', function () {
-      const lvs = document.createElement('lit-virtualizer');
+    it('registers lit-virtualizer as a custom element', async () => {
```
[the remark this answers is not in the document]
Comment (usergenic) [reply packages/labs/virtualizer/src/test/scenarios/smoke.test.ts:17]: I don't think I meant to hijack the test that only looked at the registration... I added that test back in, updated the one other one to say virtualizer items rendered in light dom.

File: packages/labs/virtualizer/src/virtualize.ts [hunk 4]
```diff
@@ -103,7 +103,7 @@ class VirtualizeDirective<T> extends AsyncDirective {
       }
       this.cachedConfig = config;
     }
-    return noChange;
+    return this.render();
     // super.update(part, [config]);
```
[the remark this answers is not in the document]
Comment (usergenic) [reply packages/labs/virtualizer/src/virtualize.ts]: Removed.

File: .changeset/yellow-dots-fix.md (hunk 1 above)
Comment (usergenic) [reply .changeset/yellow-dots-fix.md]: That reads much better.  Updated changeset.

**Review by graynorton — APPROVED** [review]
LGTM!
